test/model/findFirst.test.ts

```diff
@@ -90,3 +90,59 @@ test('returns null when found no matching entities', () => {
   })
   expect(user).toBeNull()
 })
+
+// See: https://github.com/mswjs/data/issues/78
+test('returns same results in setTimeout', async () => {
+  const db = factory({
+    user: {
+      id: primaryKey(datatype.uuid),
+      firstName: String,
+    },
+    userObject: {
+      id: primaryKey(datatype.uuid),
+      data: String,
+      user: oneOf('user'),
+    },
+  })
+
+  // Seed the database
+  const seededUser = db.user.create({
+    firstName: 'Test',
+  })
+  const seededObject = db.userObject.create({
+    user: seededUser,
+    data: 'test data - associated with user',
+  })
+
+  const queryObject = () => {
+    const object = db.userObject.findFirst({
+      where: {
+        user: {
+          id: {
+            equals: seededUser.id,
+          },
+        },
+      },
+    })
+    expect(object).toEqual({
+      data: 'test data - associated with user',
+      id: seededObject.id,
+      user: {
+        id: seededUser.id,
+        firstName: 'Test',
+      },
+    })
+  }
+
+  // Query for the object immediately
+  queryObject()
+
+  // This should return the exact same results as the query above,
+  // since nothing has changed
+  await new Promise<void>((res) =>
+    setTimeout(() => {
```

Comment:
While this test showcases the usage example of when the issue was discovered, it doesn't reflect the intention behind the test. I think the intention is for the relational properties to be enumerable. There may be multiple use cases derived from that intention, just like you've added here.

Instead, we should add a test for `defineRelationalProperties.ts` that asserts that the relational properties defined on the entity object are enumerable (huh, we don't have any tests for that function!).

Reply:
@kettanaito Good point, I can look into writing some tests for `defineRelationalProperties` then. I suppose these tests are useful in that they validate the end-to-end behavior, but do you think we should just replace theses tests with `defineRelationalProperties` since that was the root cause?

Reply:
Thank you! A single test that checks for the properties being enumerable is a good start, we can merge it at that state and iterate on it after. 

I think there's no need for the e2e test in this case, as there are no more systems involved around relational properties. If they are enumerable, you'll handle them as any other property. We may revisit this, but for now I'd say it's not necessary.